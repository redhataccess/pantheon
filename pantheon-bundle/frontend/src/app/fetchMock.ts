if (!self.fetch) {
    self.fetch = (input: RequestInfo, init?: RequestInit | undefined) => {
        const p = new Promise<Response>((resolve, reject) => {
            // just don't do anything. this exists so that our tests don't try and actually resolve fetches,
            // (which is exactly what isomorphic-fetch tried to do while we were importing it...)
            // and calling nothing here allows us to skip all of our fetch.then() and fetch.catch() code.
        })
        return p
    }
}