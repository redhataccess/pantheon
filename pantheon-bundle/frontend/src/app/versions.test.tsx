import React from "react"
import { Versions, IProps } from "@app/versions"
import "@app/fetchMock"

import { mount, shallow } from "enzyme"
import {
    Button, Form, FormGroup, FormSelect, FormSelectOption, InputGroup,
    InputGroupText, Modal, Title, Alert, AlertActionCloseButton, Grid
} from "@patternfly/react-core"
import renderer from "react-test-renderer"
import sinon from "sinon"
import { any } from "prop-types";

const anymatch = require("anymatch")

const props = {
    assemblies: [],
    attributesFilePath: "/repositories/testRepo/attributes.adoc",
    contentType: "module",
    modulePath: "/modules/test",
    onGetProduct: (productValue) => anymatch,
    onGetVersion: (versionValue) => anymatch,
    onGetUrl: (url) => anymatch,
    productInfo: "Red Hat Enterprise Linux",
    updateDate: (releaseVersion, variantUUID) => anymatch,
    variant: "test",
    variantUUID: "abcd-1234",
    versionModulePath: "/modules/test_module/en_US/variants/test/draft",
    canRegeneratePortalUrl: (regeneratePortalUrl) => anymatch
}

describe("Versions tests", () => {
    test("should render Versions component", () => {
        const view = shallow(<Versions {...props} />)
        expect(view).toMatchSnapshot()
    })

    it("should render a Modal", () => {
        const wrapper = mount(<Versions {...props} />)
        // console.log(wrapper.debug())
        const modal = wrapper.find(Modal)
        expect(modal.exists()).toBe(true)
    })

    it("should render a Grid", () => {
        const wrapper = mount(<Versions {...props} />)
        const grid = wrapper.find(Grid)
        expect(grid.exists()).toBe(true)
    })

    it("should render a Form", () => {
        const wrapper = shallow(<Versions {...props} />)
        wrapper.setState({ "login": true })
        wrapper.setState({ "isModalOpen": true })
        const form = wrapper.find(Form)
        expect(form.exists()).toBe(true)
    })

    it("should render a FormGroup", () => {
        const wrapper = shallow(<Versions {...props} />)
        wrapper.setState({ "login": true })
        wrapper.setState({ "isModalOpen": true })
        const formGroup = wrapper.find(FormGroup)
        expect(formGroup.exists()).toBe(true)
    })

    it("should render a FormSelect", () => {
        const wrapper = shallow(<Versions {...props} />)
        wrapper.setState({ "login": true })
        wrapper.setState({ "isModalOpen": true })
        const formSelect = wrapper.find(FormSelect)
        expect(formSelect.exists()).toBe(true)
    })

    it("should render a FormSelectOption", () => {
        const wrapper = shallow(<Versions {...props} />)
        wrapper.setState({ "login": true })
        wrapper.setState({ "isModalOpen": true })
        const formSelectOption = wrapper.find(FormSelectOption)
        expect(formSelectOption.exists()).toBe(true)
    })

    it("should render a InputGroup", () => {
        const wrapper = shallow(<Versions {...props} />)
        wrapper.setState({ "login": true })
        wrapper.setState({ "isModalOpen": true })
        const inputGroup = wrapper.find(InputGroup)
        expect(inputGroup.exists()).toBe(true)
    })

    it("should render a success Alert", () => {
        const wrapper = shallow(<Versions {...props} />)
        wrapper.setState({ "login": true })
        wrapper.setState({ "successAlertVisible": true })
        const alert = wrapper.find(Alert)
        expect(alert.exists()).toBe(true)
    })

    it("test fetchVersions function", () => {
        const wrapper = renderer.create(<Versions {...props} />)
        const inst = wrapper.getInstance()
        expect(inst.fetchVersions()).toMatchSnapshot()
    })

    it("test changePublishState function", () => {
        const wrapper = renderer.create(<Versions {...props} />)
        const inst = wrapper.getInstance()
        expect(inst.changePublishState("Publish")).toMatchSnapshot()
    })

    it("test previewDoc function", () => {
        const wrapper = renderer.create(<Versions {...props} />)
        const inst = wrapper.getInstance()
        expect(inst.previewDoc()).toMatchSnapshot()
    })

    it("test saveMetadata function", () => {
        const wrapper = renderer.create(<Versions {...props} />)
        const inst = wrapper.getInstance()
        expect(inst.saveMetadata()).toMatchSnapshot()
    })

    it("test onChangeUsecase function", () => {
        const wrapper = renderer.create(<Versions {...props} />)
        const inst = wrapper.getInstance()
        expect(inst.onChangeUsecase()).toMatchSnapshot()
    })

    it("test handleURLInput function", () => {
        const wrapper = renderer.create(<Versions {...props} />)
        const inst = wrapper.getInstance()
        expect(inst.handleURLInput()).toMatchSnapshot()
    })

    it("test dismissNotification function", () => {
        const wrapper = renderer.create(<Versions {...props} />)
        const inst = wrapper.getInstance()
        expect(inst.dismissNotification()).toMatchSnapshot()
    })

    it("test getMetadata function", () => {
        const wrapper = shallow(<Versions {...props} />)
        const instance = wrapper.instance()
        wrapper.setState({ "login": true })
        wrapper.setState({ "isModalOpen": true })
        // Assuming metadata exists
        const spy = sinon.spy(instance, "saveMetadata")
        const urlFragment = wrapper.find("input")
        expect(urlFragment.exists()).toBe(true)

        const useCaseValue = wrapper.find("[aria-label='FormSelect Usecase']").simulate("change", { target: { value: "Administer" } })
        expect(useCaseValue.exists()).toBe(true)
    })

    it("test getHarrayChildNamed function", () => {
        const wrapper = renderer.create(<Versions {...props} />)
        const inst = wrapper.getInstance()
        expect(inst.getHarrayChildNamed("__children__")).toMatchSnapshot()
    })

    it("test hideSuccessAlert function", () => {
        const wrapper = renderer.create(<Versions {...props} />)
        const inst = wrapper.getInstance()
        expect(inst.hideSuccessAlert()).toMatchSnapshot()
    })

    it("test hidePublishAlert function", () => {
        const wrapper = renderer.create(<Versions {...props} />);
        const inst = wrapper.getInstance();
        expect(inst.hidePublishAlert()).toMatchSnapshot();
    });

    it("test hideUppublishAlertForModule function", () => {
        const wrapper = renderer.create(<Versions {...props} />)
        const inst = wrapper.getInstance()
        expect(inst.hideUppublishAlertForModule()).toMatchSnapshot()
    })

    it("has a props", () => {
        const versions = mount(<Versions {...props} />).matchesElement
        expect(versions.length === 1)
    })

    it("test getHarrayChildNamed function", () => {
        const wrapper = renderer.create(<Versions {...props} />)
        const inst = wrapper.getInstance()
        expect(inst.getHarrayChildNamed(anymatch, "metadata")).toMatchSnapshot()
    })

    it("has a variantUUID of '1234'", () => {
        const state: IProps = {
            attributesFilePath: "/repositories/testRepo/attributes.adoc",
            contentType: "module",
            modulePath: "somePath",
            onGetProduct: (productValue) => anymatch,
            onGetVersion: (versionValue) => anymatch,
            onGetUrl: (url) => anymatch,
            productInfo: "Red Hat Enterprise Linux",
            updateDate: (releaseVersion, variantUUID) => anymatch,
            variant: "DEFAULT",
            variantUUID: "abcd-1234",
            versionModulePath: "versionPath",
            canRegeneratePortalUrl:(regeneratePortalUrl) => anymatch
        }
        state.updateDate(1, "1234")
        expect(state.modulePath).toEqual("somePath")
        expect(state.versionModulePath).toEqual("versionPath")
    })

    test("changePublishState click Publish", () => {
        const wrapper = mount(<Versions {...props} />)
        const instance = wrapper.instance()
        wrapper.setState({
            "login": true,
            "showMetadataAlertIcon": false,
            "results": [[{ "type": "draft", "icon": "BlankImage", "path": "/modules/test", "version": "Version 1", "publishedState": "Not published", "updatedDate": "", "firstButtonType": "primary", "secondButtonType": "secondary", "firstButtonText": "Publish", "secondButtonText": "Preview", "isDropdownOpen": false, "isArchiveDropDownOpen": false, "metadata": { productVersion: { label: "test", uuid: 1234 } }, "validation":[] }]],
        })
        const spy = sinon.spy(instance, "changePublishState")
        wrapper.find(Button).at(2).simulate("click")
        sinon.assert.called(spy)
    })

    test("changePublishState click Unpublish", () => {
        const wrapper = mount(<Versions {...props} />)
        const instance = wrapper.instance()
        wrapper.setState({
            "login": true,
            "results": [[{ "type": "release", "icon": "BlankImage", "path": "/modules/test", "version": "Version 1", "publishedState": "Released", "updatedDate": "", "firstButtonType": "primary", "secondButtonType": "secondary", "firstButtonText": "Publish", "secondButtonText": "Preview", "isDropdownOpen": false, "isArchiveDropDownOpen": false, "metadata": { productVersion: { label: "test", uuid: 1234 } }, "validation":[] }]],
        })
        const spy = sinon.spy(instance, "changePublishState")
        wrapper.find(Button).at(2).simulate("click")
        sinon.assert.called(spy)
    })

    it("test fetchProducts function", () => {
        const wrapper = renderer.create(<Versions {...props} />)
        const inst = wrapper.getInstance()
        const spy = sinon.spy(inst, "fetchProducts")
        inst.componentDidMount()
        sinon.assert.called(spy)
    })

    it("test fetchVersions function", () => {
        const wrapper = renderer.create(<Versions {...props} />)
        const inst = wrapper.getInstance()
        const spy = sinon.spy(inst, "fetchVersions")
        inst.fetchVersions()
        sinon.assert.called(spy)
    })

    it("test changePublishState function", () => {
        const wrapper = renderer.create(<Versions {...props} />)
        const inst = wrapper.getInstance()
        const spy = sinon.spy(inst, "changePublishState")
        inst.changePublishState("ds")
        sinon.assert.called(spy)
    })

    it("test getDocumentsIncluded function", () => {
        const wrapper = renderer.create(<Versions {...props} />)
        const inst = wrapper.getInstance()
        const spy = sinon.spy(inst, "getDocumentsIncluded")
        inst.getDocumentsIncluded()
        sinon.assert.called(spy)
    })

    it("test capitalize function", () => {
        const wrapper = renderer.create(<Versions {...props} />)
        const inst = wrapper.getInstance()
        const spy = sinon.spy(inst, "capitalize")
        inst.capitalize()
        sinon.assert.called(spy)
    })

    it("test setAlertTitle function", () => {
        const wrapper = renderer.create(<Versions {...props} />)
        const inst = wrapper.getInstance()
        const spy = sinon.spy(inst, "setAlertTitle")
        inst.setAlertTitle()
        sinon.assert.called(spy)
    })
})
