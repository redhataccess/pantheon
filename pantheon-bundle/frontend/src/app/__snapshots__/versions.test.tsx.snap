// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Versions tests should render Versions component 1`] = `
<Fragment>
  <Card>
    <div>
      <DataList
        aria-label="Simple data list"
      >
        <DataListItem
          aria-labelledby="simple-item1"
          isExpanded={true}
        >
          <DataListItemRow
            id="data-rows-header"
          >
            <DataListToggle
              aria-controls="width-ex3-expand1"
              id="width-ex3-toggle1"
              isExpanded={true}
              onClick={[Function]}
            />
            <DataListItemCells
              dataListCells={
                Array [
                  <DataListCell>
                    <span
                      className="sp-prop-nosort"
                      id="span-source-type-version"
                    >
                      Version
                    </span>
                  </DataListCell>,
                  <DataListCell>
                    <span
                      className="sp-prop-nosort"
                      id="span-source-type-version-published"
                    >
                      Published
                    </span>
                  </DataListCell>,
                  <DataListCell>
                    <span
                      className="sp-prop-nosort"
                      id="span-source-type-version-draft-uploaded"
                    >
                      Draft Uploaded
                    </span>
                  </DataListCell>,
                  <DataListCell>
                    <span
                      className="sp-prop-nosort"
                      id="span-source-name-version-publish-buttons"
                    />
                  </DataListCell>,
                  <DataListCell>
                    <span
                      className="sp-prop-nosort"
                      id="span-source-name"
                    />
                  </DataListCell>,
                ]
              }
            />
          </DataListItemRow>
        </DataListItem>
        <DataListContent
          aria-label="Secondary Content Details"
          id="Content"
          isHidden={false}
          key="details_dlc"
          noPadding={true}
        />
      </DataList>
    </div>
  </Card>
  <Modal
    actions={
      Array [
        <Unknown
          form="edit_metadata"
          onClick={[Function]}
          variant="primary"
        >
          Save
        </Unknown>,
        <Unknown
          onClick={[Function]}
          variant="secondary"
        >
          Cancel
        </Unknown>,
      ]
    }
    appendTo={<body />}
    ariaDescribedById="edit-metadata"
    className=""
    header={
      <React.Fragment>
        <Title
          headingLevel="h1"
          size="2xl"
        >
          Edit Metadata
        </Title>
        <br />
        <p
          className="pf-u-pl-sm"
        >
          All fields are required.
        </p>
      </React.Fragment>
    }
    hideTitle={false}
    isFooterLeftAligned={false}
    isLarge={false}
    isOpen={false}
    isSmall={false}
    onClose={[Function]}
    showClose={true}
    title="Edit metadata"
    width="60%"
  >
    <div />
    <div
      className="app-container"
    />
    <Form
      id="edit_metadata"
      isHorizontal={true}
    >
      <FormGroup
        fieldId="product-name"
        isRequired={true}
        label="Product Name"
      >
        <InputGroup>
          <FormSelect
            aria-label="FormSelect Product"
            className=""
            isDisabled={false}
            isRequired={false}
            isValid={true}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            value=""
          >
            <FormSelectOption
              label="Select a Product"
            />
          </FormSelect>
          <FormSelect
            aria-label="FormSelect Version"
            className=""
            id="productVersion"
            isDisabled={false}
            isRequired={false}
            isValid={true}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            value=""
          >
            <FormSelectOption
              label="Select a Version"
            />
          </FormSelect>
        </InputGroup>
      </FormGroup>
      <FormGroup
        fieldId="document-usecase"
        isRequired={true}
        label="Document use case"
      >
        <FormSelect
          aria-label="FormSelect Usecase"
          className=""
          isDisabled={false}
          isRequired={false}
          isValid={true}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          value=""
        >
          <FormSelectOption
            key="usecase_0"
            label="Select Use Case"
            value="Select Use Case"
          />
          <FormSelectOption
            key="usecase_1"
            label="Administer"
            value="Administer"
          />
          <FormSelectOption
            key="usecase_2"
            label="Deploy"
            value="Deploy"
          />
          <FormSelectOption
            key="usecase_3"
            label="Develop"
            value="Develop"
          />
          <FormSelectOption
            key="usecase_4"
            label="Install"
            value="Install"
          />
          <FormSelectOption
            key="usecase_5"
            label="Migrate"
            value="Migrate"
          />
          <FormSelectOption
            key="usecase_6"
            label="Monitor"
            value="Monitor"
          />
          <FormSelectOption
            key="usecase_7"
            label="Network"
            value="Network"
          />
          <FormSelectOption
            key="usecase_8"
            label="Plan"
            value="Plan"
          />
          <FormSelectOption
            key="usecase_9"
            label="Provision"
            value="Provision"
          />
          <FormSelectOption
            key="usecase_10"
            label="Release"
            value="Release"
          />
          <FormSelectOption
            key="usecase_11"
            label="Troubleshoot"
            value="Troubleshoot"
          />
          <FormSelectOption
            key="usecase_12"
            label="Optimize"
            value="Optimize"
          />
        </FormSelect>
      </FormGroup>
      <FormGroup
        fieldId="url-fragment"
        isRequired={true}
        label="Vanity URL fragment"
      >
        <InputGroup>
          <InputGroupText
            aria-label="/"
            id="slash"
          >
            <span>
              /
            </span>
          </InputGroupText>
          <TextInput
            aria-label={null}
            className=""
            id="url-fragment"
            isDisabled={false}
            isReadOnly={false}
            isRequired={true}
            isValid={true}
            onChange={[Function]}
            placeholder="Enter URL"
            type="text"
            value=""
          />
        </InputGroup>
      </FormGroup>
      <FormGroup
        fieldId="search-keywords"
        isRequired={false}
        label="Search keywords"
      >
        <InputGroup>
          <TextInput
            aria-label={null}
            className=""
            id="search-keywords"
            isDisabled={false}
            isReadOnly={false}
            isRequired={false}
            isValid={true}
            onChange={[Function]}
            placeholder="cat, dog, bird..."
            type="text"
            value=""
          />
        </InputGroup>
      </FormGroup>
      <div>
        <input
          name="productVersion@TypeHint"
          type="hidden"
          value="Reference"
        />
      </div>
    </Form>
  </Modal>
</Fragment>
`;

exports[`Versions tests test changePublishState function 1`] = `undefined`;

exports[`Versions tests test dismissNotification function 1`] = `undefined`;

exports[`Versions tests test fetchVersions function 1`] = `undefined`;

exports[`Versions tests test getHarrayChildNamed function 1`] = `""`;

exports[`Versions tests test getHarrayChildNamed function 2`] = `""`;

exports[`Versions tests test handleURLInput function 1`] = `undefined`;

exports[`Versions tests test hidePublishAlert function 1`] = `undefined`;

exports[`Versions tests test hideSuccessAlert function 1`] = `undefined`;

exports[`Versions tests test loginRedirect function 1`] = `""`;

exports[`Versions tests test onArchiveSelect function 1`] = `undefined`;

exports[`Versions tests test onArchiveToggle function 1`] = `undefined`;

exports[`Versions tests test onChangeUsecase function 1`] = `undefined`;

exports[`Versions tests test onExpandableToggle function 1`] = `undefined`;

exports[`Versions tests test onHeadingToggle function 1`] = `undefined`;

exports[`Versions tests test previewDoc function 1`] = `undefined`;

exports[`Versions tests test saveMetadata function 1`] = `undefined`;
